installer/pkg/workflow/destroy.go

```diff
@@ -30,7 +43,24 @@ func destroyTopologyStep(m *metadata) error {
 }
 
 func destroyJoinWorkersStep(m *metadata) error {
-	return runDestroyStep(m, joinWorkersStep)
+	deleteWorkerMachineSet(filepath.Join(m.clusterDir, kubeconfigPath))
+	return nil
+}
+
+func deleteWorkerMachineSet(kubeconfig string) error {
```

Comment:
This doesn't perform validation if the worker machineset was actually scaled to 0